core/l10n_init.js

```diff
@@ -61,6 +61,8 @@ define('core/l10n_init',
     }
 
     function _transformLocale(locale) {
+        // Use default locale "en-US" if given locale is undefined (BUG 1138558)
+        locale = locale || 'en-US';
```

Comment:
Nah no offense, I think this is totally awesome. I don't think my nitpicks are to worry about. Let's just go with it.